x/community/client/utils/utils.go

```diff
@@ -37,3 +38,48 @@ func ParseCommunityPoolLendWithdrawProposal(
 	err = cdc.UnmarshalJSON(contents, &proposal)
 	return proposal, err
 }
+
+// communityPoolJSONProposal defines the new Msg-based proposal.
+type communityPoolJSONProposal struct {
+	Title       string            `json:"title"`
+	Description string            `json:"description"`
+	Messages    []json.RawMessage `json:"messages,omitempty"`
+	Deposit     string            `json:"deposit"`
+}
+
+// ParseCommunityPoolProposalJSON reads and parses a CommunityPoolProposalJSON from a file.
+func ParseCommunityPoolProposalJSON(cdc codec.JSONCodec, proposalFile string) (*types.CommunityPoolProposal, sdk.Coins, error) {
+	var jsonProposal communityPoolJSONProposal
+	contents, err := os.ReadFile(proposalFile)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	err = json.Unmarshal(contents, &jsonProposal)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	msgs := make([]sdk.Msg, len(jsonProposal.Messages))
+	for i, anyJSON := range jsonProposal.Messages {
+		var msg sdk.Msg
+		err := cdc.UnmarshalInterfaceJSON(anyJSON, &msg)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		msgs[i] = msg
+	}
+
+	deposit, err := sdk.ParseCoinsNormalized(jsonProposal.Deposit)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	communityPoolProposal, err := types.NewCommunityPoolProposal(jsonProposal.Title, jsonProposal.Description, msgs)
```

Comment:
for best user experience / principle of least surprise recommend validating at proposal time here that the message not only parses - is syntactially correct, but also is semantically correct -e.g. that the authority is in the list of whitelisted addresses, and that the type of the message to be called is in the whitelist of messages that the community pool is authorized to execute.